--- spack-src/source/makefile.orig	2019-12-02 14:12:58.000000000 +0900
+++ spack-src/source/makefile	2021-03-17 13:39:00.000000000 +0900
@@ -149,20 +149,20 @@
 #-----------------------------------------------------------------------
 # LIBELPA
 #-----------------------------------------------------------------------
-LIBELPADIR = ./elpa-2018.05.001
+#LIBELPADIR = ./elpa-2018.05.001
 
-OBJS    += mod_precision.o elpa_utilities.o\
-	elpa1_compute_real.o elpa1_compute_complex.o\
-	aligned_mem.o elpa2_determine_workload.o\
-	mod_redist_band_real.o mod_redist_band_complex.o\
-	mod_pack_unpack_cpu_real.o mod_pack_unpack_cpu_complex.o\
-	real.o complex.o\
-	mod_single_hh_trafo_real.o mod_compute_hh_trafo_real.o mod_compute_hh_trafo_complex.o\
-	elpa2_compute_real.o elpa2_compute_complex.o \
-	elpa_solve_evp_real_2stage_double_impl.o elpa_solve_evp_complex_2stage_double_impl.o\
+#OBJS    += mod_precision.o elpa_utilities.o\
+#	elpa1_compute_real.o elpa1_compute_complex.o\
+#	aligned_mem.o elpa2_determine_workload.o\
+#	mod_redist_band_real.o mod_redist_band_complex.o\
+#	mod_pack_unpack_cpu_real.o mod_pack_unpack_cpu_complex.o\
+#	real.o complex.o\
+#	mod_single_hh_trafo_real.o mod_compute_hh_trafo_real.o mod_compute_hh_trafo_complex.o\
+#	elpa2_compute_real.o elpa2_compute_complex.o \
+#	elpa_solve_evp_real_2stage_double_impl.o elpa_solve_evp_complex_2stage_double_impl.o\
 
-CC      += -I$(LIBELPADIR)
-FC      += -I$(LIBELPADIR)
+#CC      += -I$(LIBELPADIR)
+#FC      += -I$(LIBELPADIR)
 
 #
 # set program name
@@ -953,44 +953,44 @@
 solve_evp_complex.o: solve_evp_complex.f90 elpa1.o
 	$(FC) -c solve_evp_complex.f90
 
-mod_precision.o: $(LIBELPADIR)/mod_precision.F90
-	$(FC) -c $(LIBELPADIR)/mod_precision.F90
-elpa_utilities.o: $(LIBELPADIR)/elpa_utilities.F90
-	$(FC) -c $(LIBELPADIR)/elpa_utilities.F90
-elpa1_compute_real.o: $(LIBELPADIR)/elpa1_compute_real.F90
-	$(FC) -c $(LIBELPADIR)/elpa1_compute_real.F90
-elpa1_compute_complex.o: $(LIBELPADIR)/elpa1_compute_complex.F90
-	$(FC) -c $(LIBELPADIR)/elpa1_compute_complex.F90
-aligned_mem.o: $(LIBELPADIR)/aligned_mem.F90
-	$(FC) -c $(LIBELPADIR)/aligned_mem.F90
-mod_pack_unpack_cpu_real.o: $(LIBELPADIR)/mod_pack_unpack_cpu_real.F90
-	$(FC) -c $(LIBELPADIR)/mod_pack_unpack_cpu_real.F90
-mod_pack_unpack_cpu_complex.o: $(LIBELPADIR)/mod_pack_unpack_cpu_complex.F90
-	$(FC) -c $(LIBELPADIR)/mod_pack_unpack_cpu_complex.F90
-mod_redist_band_real.o: $(LIBELPADIR)/mod_redist_band_real.F90
-	$(FC) -c $(LIBELPADIR)/mod_redist_band_real.F90
-mod_redist_band_complex.o: $(LIBELPADIR)/mod_redist_band_complex.F90
-	$(FC) -c $(LIBELPADIR)/mod_redist_band_complex.F90
-elpa2_determine_workload.o: $(LIBELPADIR)/elpa2_determine_workload.F90
-	$(FC) -c $(LIBELPADIR)/elpa2_determine_workload.F90
-real.o: $(LIBELPADIR)/real.F90
-	$(FC) -c $(LIBELPADIR)/real.F90
-complex.o: $(LIBELPADIR)/complex.F90
-	$(FC) -c $(LIBELPADIR)/complex.F90
-mod_single_hh_trafo_real.o: $(LIBELPADIR)/mod_single_hh_trafo_real.F90
-	$(FC) -c $(LIBELPADIR)/mod_single_hh_trafo_real.F90
-mod_compute_hh_trafo_real.o: $(LIBELPADIR)/mod_compute_hh_trafo_real.F90
-	$(FC) -c $(LIBELPADIR)/mod_compute_hh_trafo_real.F90
-mod_compute_hh_trafo_complex.o: $(LIBELPADIR)/mod_compute_hh_trafo_complex.F90
-	$(FC) -c $(LIBELPADIR)/mod_compute_hh_trafo_complex.F90
-elpa2_compute_real.o: $(LIBELPADIR)/elpa2_compute_real.F90
-	$(FC) -c $(LIBELPADIR)/elpa2_compute_real.F90
-elpa2_compute_complex.o: $(LIBELPADIR)/elpa2_compute_complex.F90
-	$(FC) -c $(LIBELPADIR)/elpa2_compute_complex.F90
-elpa_solve_evp_real_2stage_double_impl.o: $(LIBELPADIR)/elpa_solve_evp_real_2stage_double_impl.F90
-	$(FC) -c $(LIBELPADIR)/elpa_solve_evp_real_2stage_double_impl.F90
-elpa_solve_evp_complex_2stage_double_impl.o: $(LIBELPADIR)/elpa_solve_evp_complex_2stage_double_impl.F90
-	$(FC) -c $(LIBELPADIR)/elpa_solve_evp_complex_2stage_double_impl.F90
+#mod_precision.o: $(LIBELPADIR)/mod_precision.F90
+#	$(FC) -c $(LIBELPADIR)/mod_precision.F90
+#elpa_utilities.o: $(LIBELPADIR)/elpa_utilities.F90
+#	$(FC) -c $(LIBELPADIR)/elpa_utilities.F90
+#elpa1_compute_real.o: $(LIBELPADIR)/elpa1_compute_real.F90
+#	$(FC) -c $(LIBELPADIR)/elpa1_compute_real.F90
+#elpa1_compute_complex.o: $(LIBELPADIR)/elpa1_compute_complex.F90
+#	$(FC) -c $(LIBELPADIR)/elpa1_compute_complex.F90
+#aligned_mem.o: $(LIBELPADIR)/aligned_mem.F90
+#	$(FC) -c $(LIBELPADIR)/aligned_mem.F90
+#mod_pack_unpack_cpu_real.o: $(LIBELPADIR)/mod_pack_unpack_cpu_real.F90
+#	$(FC) -c $(LIBELPADIR)/mod_pack_unpack_cpu_real.F90
+#mod_pack_unpack_cpu_complex.o: $(LIBELPADIR)/mod_pack_unpack_cpu_complex.F90
+#	$(FC) -c $(LIBELPADIR)/mod_pack_unpack_cpu_complex.F90
+#mod_redist_band_real.o: $(LIBELPADIR)/mod_redist_band_real.F90
+#	$(FC) -c $(LIBELPADIR)/mod_redist_band_real.F90
+#mod_redist_band_complex.o: $(LIBELPADIR)/mod_redist_band_complex.F90
+#	$(FC) -c $(LIBELPADIR)/mod_redist_band_complex.F90
+#elpa2_determine_workload.o: $(LIBELPADIR)/elpa2_determine_workload.F90
+#	$(FC) -c $(LIBELPADIR)/elpa2_determine_workload.F90
+#real.o: $(LIBELPADIR)/real.F90
+#	$(FC) -c $(LIBELPADIR)/real.F90
+#complex.o: $(LIBELPADIR)/complex.F90
+#	$(FC) -c $(LIBELPADIR)/complex.F90
+#mod_single_hh_trafo_real.o: $(LIBELPADIR)/mod_single_hh_trafo_real.F90
+#	$(FC) -c $(LIBELPADIR)/mod_single_hh_trafo_real.F90
+#mod_compute_hh_trafo_real.o: $(LIBELPADIR)/mod_compute_hh_trafo_real.F90
+#	$(FC) -c $(LIBELPADIR)/mod_compute_hh_trafo_real.F90
+#mod_compute_hh_trafo_complex.o: $(LIBELPADIR)/mod_compute_hh_trafo_complex.F90
+#	$(FC) -c $(LIBELPADIR)/mod_compute_hh_trafo_complex.F90
+#elpa2_compute_real.o: $(LIBELPADIR)/elpa2_compute_real.F90
+#	$(FC) -c $(LIBELPADIR)/elpa2_compute_real.F90
+#elpa2_compute_complex.o: $(LIBELPADIR)/elpa2_compute_complex.F90
+#	$(FC) -c $(LIBELPADIR)/elpa2_compute_complex.F90
+#elpa_solve_evp_real_2stage_double_impl.o: $(LIBELPADIR)/elpa_solve_evp_real_2stage_double_impl.F90
+#	$(FC) -c $(LIBELPADIR)/elpa_solve_evp_real_2stage_double_impl.F90
+#elpa_solve_evp_complex_2stage_double_impl.o: $(LIBELPADIR)/elpa_solve_evp_complex_2stage_double_impl.F90
+#	$(FC) -c $(LIBELPADIR)/elpa_solve_evp_complex_2stage_double_impl.F90
 
 
 #
