diff -u -r --no-dereference a/cpp/cmake_modules/SetupCxxFlags.cmake b/cpp/cmake_modules/SetupCxxFlags.cmake
--- a/cpp/cmake_modules/SetupCxxFlags.cmake	2021-01-19 06:42:56.000000000 +0900
+++ b/cpp/cmake_modules/SetupCxxFlags.cmake	2023-01-30 23:01:42.000000000 +0900
@@ -304,6 +304,7 @@
     set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} /W3")
   elseif(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang"
          OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang"
+         OR CMAKE_CXX_COMPILER_ID STREQUAL "FujitsuClang"
          OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU"
          OR CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
     set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} -Wall")
