diff -u -r a/src/muz/spacer/spacer_arith_kernel.cpp b/src/muz/spacer/spacer_arith_kernel.cpp
--- a/src/muz/spacer/spacer_arith_kernel.cpp	2022-09-04 07:47:12.000000000 +0900
+++ b/src/muz/spacer/spacer_arith_kernel.cpp	2023-02-16 15:50:55.000000000 +0900
@@ -75,7 +75,9 @@
         for (auto row : kern.get_rows()) {
             vec.reset();
             vec.reserve(kern.num_vars(), rational(0));
-            for (auto &[coeff, v] : kern.get_row(row)) {
+            for (auto &e : kern.get_row(row)) {
+                auto &coeff = e.m_coeff;
+                auto &v = e.m_var;
                 vec[v] = rational(coeff);
             }
             out.add_row(vec);
