diff -ru orig/siesta-MaX-1.0.1/Src/basis_io.F siesta-MaX-1.0.1/Src/basis_io.F
--- orig/siesta-MaX-1.0.1/Src/basis_io.F	2020-02-07 20:40:31.000000000 +0900
+++ siesta-MaX-1.0.1/Src/basis_io.F	2021-10-28 10:20:51.000000000 +0900
@@ -743,7 +743,12 @@
          call check(iret)
          iret = nf90_put_var(ncid,pjnl_j_id,spp%pjnl_j,count=(/nkbs/))
          call check(iret)
+         write(*,*) 'ncid = ', ncid
+         write(*,*) 'pjnl_n_id = ', pjnl_n_id
+         write(*,*) 'spp%pjnl_n= ', spp%pjnl_n
+         write(*,*) 'nkbs= ', nkbs 
          iret = nf90_put_var(ncid,pjnl_n_id,spp%pjnl_n,count=(/nkbs/))
+         write(*,*) 'iret = ', iret
          call check(iret)
          iret = nf90_put_var(ncid,pjnl_ekb_id,spp%pjnl_ekb,
      $        count=(/nkbs/))
