--- orig/siesta-MaX-1.0.1/Util/build_all.sh	2020-02-07 20:40:31.000000000 +0900
+++ siesta-MaX-1.0.1/Util/build_all.sh	2021-09-04 00:04:44.000000000 +0900
@@ -37,14 +37,14 @@
 	    continue
 	    ;;
     esac
-    
+
     echo ""
     echo "====> Processing $relpath ..."
     echo ""
     cd $relpath
     
-    make OBJDIR=${OBJDIR} clean 
-    make OBJDIR=${OBJDIR}
+    make OBJDIR=${OBJDIR} clean > /dev/null 2>&1
+    make OBJDIR=${OBJDIR} > /dev/null 2>&1
     if [ $? -ne 0 ]; then
 	echo "*** COMPILATION FAILED in $relpath ***"
 	failed="$failed $relpath"
