--- spack-src/test/testTECalc.py.orig	2020-05-25 16:11:28.000000000 +0900
+++ spack-src/test/testTECalc.py	2020-11-28 12:32:49.000000000 +0900
@@ -208,7 +208,7 @@
 
     sin=StandardIni(Model=Model)
     MakeStandard(sin, "stan.in")
-    subprocess.call(PathToHPhi+" -sdry ./stan.in", shell=True)
+    subprocess.call("mpiexec -np 1 "+PathToHPhi+" -sdry ./stan.in", shell=True)
 
     #[s] Make CalcMod file
     #For Lanczos
