Only in libgridxc-0.8.5: build
diff -ru orig/libgridxc-0.8.5/src/sorting.F90 libgridxc-0.8.5/src/sorting.F90
--- orig/libgridxc-0.8.5/src/sorting.F90	2019-02-13 22:48:02.000000000 +0900
+++ libgridxc-0.8.5/src/sorting.F90	2021-08-16 20:06:46.039238991 +0900
@@ -189,6 +189,7 @@
     ! *************** INPUT AND OUTPUT **********************************
     ! REAL*8  X(M,N) : Array(s) to be ordered: Xout(I,J) = Xin(I,INDEX(J))
     ! *******************************************************************
+    use precision, only: dp
     INTEGER, intent(in) :: M, N, INDX(N)
     REAL(DP), intent(inout) :: X(M,N)
     X = X(:,INDX)
