--- orig/libpsml-1.1.7/src/m_psml.F90	2017-11-06 22:12:13.000000000 +0900
+++ libpsml-1.1.7/src/m_psml.F90	2021-08-18 20:13:02.000000000 +0900
@@ -5,10 +5,10 @@
 module m_psml
 
 !
-  use m_psml_reader, only: psml_reader 
 
   use m_psml_core, only: ps_t, ps_destroy
 
+  use m_psml_reader, only: psml_reader 
   use m_psml_core, only: str_of_set
   use m_psml_core, only:     SET_SREL     , &
                              SET_NONREL   , &
