diff -ruN orig/blitz-1.0.1/blitz/generate/genstencils.py blitz-1.0.1/blitz/generate/genstencils.py
--- orig/blitz-1.0.1/blitz/generate/genstencils.py	2017-10-03 01:04:43.000000000 +0900
+++ blitz-1.0.1/blitz/generate/genstencils.py	2020-12-18 21:56:59.000000000 +0900
@@ -6,7 +6,7 @@
 import sys
 
 os=open(sys.argv[1],'w')
-print "Generating file %s"%sys.argv[1]
+print("Generating file %s"%(sys.argv[1]))
 
 def BZ_ET_STENCIL_REDIRECT(name):
     stub="""
