--- spack-src/tools/build/src/tools/clang-linux.jam.orig	2021-11-30 19:33:39.000000000 +0900
+++ spack-src/tools/build/src/tools/clang-linux.jam	2021-11-30 19:34:29.000000000 +0900
@@ -108,7 +108,7 @@
 
 rule get-full-version ( command-string )
 {
-    return [ common.match-command-output version : "version ([0-9.]+)"
+    return [ common.match-command-output version : "Compiler ([0-9.]+)"
            : "$(command-string) --version" ] ;
 }
 
