diff -r 46387cc3b082 etc/bashrc
--- OpenFOAM-v2106/etc/bashrc	Mon Sep 06 16:48:22 2021 +0900
+++ OpenFOAM-v2106/etc/bashrc	Mon Sep 06 16:52:19 2021 +0900
@@ -123,7 +123,7 @@
 # [FOAM_SIGFPE] - Trap floating-point exceptions.
 #               - overrides the 'trapFpe' controlDict entry
 # = true | false
-#export FOAM_SIGFPE=true
+export FOAM_SIGFPE=false
 
 # [FOAM_SETNAN] - Initialize memory with NaN
 #               - overrides the 'setNaN' controlDict entry
diff -r 46387cc3b082 wmake/rules/linuxARM64Fujitsu/c
--- OpenFOAM-v2106/wmake/rules/linuxARM64Fujitsu/c	Mon Sep 06 16:48:22 2021 +0900
+++ OpenFOAM-v2106/wmake/rules/linuxARM64Fujitsu/c	Mon Sep 06 16:52:19 2021 +0900
@@ -1,9 +1,9 @@
 # Fujitsu compiler - a clang variant
 include $(GENERAL_RULES)/Clang/c
 
-cc          = fcc
+cc          = fcc -Nclang
 
-cARCH       =
+cARCH       = -m64 -march=armv8.2-a+sve -mcpu=a64fx
 
 # Compile option is non-mandatory, but must be non-empty
 sinclude $(DEFAULT_RULES)/c$(WM_COMPILE_OPTION)
diff -r 46387cc3b082 wmake/rules/linuxARM64Fujitsu/c++
--- OpenFOAM-v2106/wmake/rules/linuxARM64Fujitsu/c++	Mon Sep 06 16:48:22 2021 +0900
+++ OpenFOAM-v2106/wmake/rules/linuxARM64Fujitsu/c++	Mon Sep 06 16:52:19 2021 +0900
@@ -1,9 +1,9 @@
 # Fujitsu compiler - a clang variant
 include $(GENERAL_RULES)/Clang/c++
 
-CC          = FCC -std=c++11
+CC          = FCC -Nclang -std=c++14
 
-c++ARCH     = -pthread
+c++ARCH     = -pthread -m64 -march=armv8.2-a+sve -mcpu=a64fx
 
 include $(DEFAULT_RULES)/c++$(WM_COMPILE_OPTION)
 
