--- spack-src/cmake/FindMumps.cmake.org	2023-12-08 19:43:27.000000000 +0900
+++ spack-src/cmake/FindMumps.cmake	2023-12-09 00:10:19.000000000 +0900
@@ -18,7 +18,7 @@
   RETURN()
 endif()
 
-if($ENV{MUMPS_ROOT})
+if(DEFINED ENV{MUMPS_ROOT})
   set(LIB_SEARCH_PATH ${LIB_SEARCH_PATH} $ENV{MUMPS_ROOT}/lib)
 endif()
 set(LIB_SEARCH_PATH ${LIB_SEARCH_PATH} ${CMAKE_SOURCE_DIR}/../MUMPS_5.3.3/lib)
@@ -39,7 +39,7 @@
 set(LIB_SEARCH_PATH ${LIB_SEARCH_PATH} /usr/lib)
 
 if(NOT WITH_MPI)
-  if($ENV{MUMPS_ROOT})
+  if(DEFINED ENV{MUMPS_ROOT})
     set(LIB_SEARCH_PATH ${LIB_SEARCH_PATH} $ENV{MUMPS_ROOT}/libseq)
   endif()
   set(LIB_SEARCH_PATH ${LIB_SEARCH_PATH} ${CMAKE_SOURCE_DIR}/../MUMPS_5.3.3/libseq)
