--- spack-src/cmake/FindRefiner.cmake.org	2021-03-18 14:50:27.000000000 +0900
+++ spack-src/cmake/FindRefiner.cmake	2021-03-18 16:09:46.000000000 +0900
@@ -22,6 +22,7 @@
 find_path(REFINER_INCLUDE_PATH
   NAMES rcapRefiner.h
   HINTS $ENV{REFINER_ROOT}/Refiner
+    $ENV{REFINER_ROOT}/include
     ${CMAKE_SOURCE_DIR}/../REVOCAP_Refiner-1.1.04/Refiner
     $ENV{HOME}/REVOCAP_Refiner-1.1.04/Refiner
     $ENV{HOME}/local/include
@@ -36,6 +37,7 @@
   HINTS $ENV{REFINER_ROOT}/lib/${CMAKE_SYSTEM_PROCESSOR}-${REFINER_SYSTEM_NAME}
     ${CMAKE_SOURCE_DIR}/../REVOCAP_Refiner-1.1.04/lib/${CMAKE_SYSTEM_PROCESSOR}-${REFINER_SYSTEM_NAME}
     $ENV{HOME}/REVOCAP_Refiner-1.1.04/lib/${CMAKE_SYSTEM_PROCESSOR}-${REFINER_SYSTEM_NAME}
+    $ENV{REFINER_ROOT}/lib
     $ENV{HOME}/local/lib
     $ENV{HOME}/.local/lib
     ${CMAKE_LIBRARY_PATH}
