--- spack-src/cmake/FindParmetis.cmake.org	2023-12-08 19:43:27.000000000 +0900
+++ spack-src/cmake/FindParmetis.cmake	2023-12-09 00:11:28.000000000 +0900
@@ -22,7 +22,7 @@
 if(NOT PARMETIS_VER_3)
   find_path(PARMETIS_INCLUDE_PATH
     NAMES parmetis.h
-    HINTS ${PARMETIS_ROOT}/include
+    HINTS $ENV{PARMETIS_ROOT}/include
     ${CMAKE_SOURCE_DIR}/../parmetis-4.0.3/include
     $ENV{HOME}/parmetis-4.0.3/include
     $ENV{HOME}/local/include
@@ -36,6 +36,7 @@
   find_library(PARMETIS_LIBRARIES
     NAMES parmetis
     HINTS $ENV{PARMETIS_ROOT}/build/${CMAKE_SYSTEM_NAME}-${CMAKE_SYSTEM_PROCESSOR}/libparmetis
+    $ENV{PARMETIS_ROOT}/lib
     ${CMAKE_SOURCE_DIR}/../parmetis-4.0.3/build/${CMAKE_SYSTEM_NAME}-${CMAKE_SYSTEM_PROCESSOR}/libparmetis
     $ENV{HOME}/parmetis-4.0.3/build/${CMAKE_SYSTEM_NAME}-${CMAKE_SYSTEM_PROCESSOR}/libparmetis
     $ENV{HOME}/local/lib
