--- spack-src/cmake/FindRevocap.cmake.org	2021-03-18 14:50:27.000000000 +0900
+++ spack-src/cmake/FindRevocap.cmake	2021-03-18 14:50:27.000000000 +0900
@@ -20,7 +20,7 @@
 
 find_path(REVOCAP_INCLUDE_PATH
   NAMES rcap.h
-  HINTS $ENV{REVOCAP_ROOT}/librcap
+  HINTS $ENV{REVOCAP_ROOT}/include
     ${CMAKE_SOURCE_DIR}/../REVOCAP_Coupler-2.1/librcap
     $ENV{HOME}/REVOCAP_Coupler-2.1/librcap
     $ENV{HOME}/local/include
@@ -33,7 +33,7 @@
 
 find_library(REVOCAP_RCAP_LIBRARY
   NAMES rcap
-  HINTS $ENV{REVOCAP_ROOT}/librcap
+  HINTS $ENV{REVOCAP_ROOT}/lib
     ${CMAKE_SOURCE_DIR}/../REVOCAP_Coupler-2.1/librcap
     $ENV{HOME}/REVOCAP_Coupler-2.1/librcap
     $ENV{HOME}/local/lib
@@ -46,7 +46,7 @@
 
 find_library(REVOCAP_RCAPF_LIBRARY
   NAMES rcapf
-  HINTS $ENV{REVOCAP_ROOT}/librcap
+  HINTS $ENV{REVOCAP_ROOT}/lib
     ${CMAKE_SOURCE_DIR}/../REVOCAP_Coupler-2.1/librcap
     $ENV{HOME}/REVOCAP_Coupler-2.1/librcap
     $ENV{HOME}/local/lib
