diff --git a/fistr1/src/common/fstr_ctrl_common.f90 b/fistr1/src/common/fstr_ctrl_common.f90
index d1b326b8..1d224940 100644
--- a/fistr1/src/common/fstr_ctrl_common.f90
+++ b/fistr1/src/common/fstr_ctrl_common.f90
@@ -448,7 +448,7 @@ contains
     use m_out
     integer(kind=kint), intent(in)        :: ctrl      !< readed data
     type (hecmwST_local_mesh), intent(in) :: hecMESH   !< mesh information
-    type( output_info ), intent(out)      :: outinfo   !< output information
+    type( output_info ), intent(inout)    :: outinfo   !< output information
 
     integer(kind=kint) :: rcode, ipos
     integer(kind=kint) :: n, i, j
