diff -u -r kokkos-3.4.01/cmake/kokkos_enable_devices.cmake kokkos-3.4.01-fj/cmake/kokkos_enable_devices.cmake
--- kokkos-3.4.01/cmake/kokkos_enable_devices.cmake	2021-05-26 00:35:14.000000000 +0900
+++ kokkos-3.4.01-fj/cmake/kokkos_enable_devices.cmake	2021-12-13 13:13:38.000000000 +0900
@@ -72,6 +72,7 @@
       Clang      ${ClangOpenMPFlag}
       IntelLLVM  -fiopenmp
       AppleClang -Xpreprocessor -fopenmp
+      FujitsuClang -Kopenmp
       PGI        -mp
       Cray       NO-VALUE-SPECIFIED
       XL         -qsmp=omp
diff -u -r kokkos-3.4.01/cmake/kokkos_functions.cmake kokkos-3.4.01-fj/cmake/kokkos_functions.cmake
--- kokkos-3.4.01/cmake/kokkos_functions.cmake	2021-05-26 00:35:14.000000000 +0900
+++ kokkos-3.4.01-fj/cmake/kokkos_functions.cmake	2021-12-13 13:11:33.000000000 +0900
@@ -773,7 +773,8 @@
 ENDFUNCTION()
 
 FUNCTION(COMPILER_SPECIFIC_OPTIONS_HELPER)
-  SET(COMPILERS NVIDIA PGI XL DEFAULT Cray Intel Clang AppleClang IntelLLVM GNU HIPCC Fujitsu)
+  #SET(COMPILERS NVIDIA PGI XL DEFAULT Cray Intel Clang AppleClang IntelLLVM GNU HIPCC Fujitsu)
+  SET(COMPILERS NVIDIA PGI XL DEFAULT Cray Intel Clang AppleClang IntelLLVM GNU HIPCC FujitsuClang Fujitsu)
   CMAKE_PARSE_ARGUMENTS(
     PARSE
     "LINK_OPTIONS;COMPILE_OPTIONS;COMPILE_DEFINITIONS;LINK_LIBRARIES"
