diff -u -r a/cmake/kokkos_enable_devices.cmake b/cmake/kokkos_enable_devices.cmake
--- a/cmake/kokkos_enable_devices.cmake	2022-09-21 07:17:28.000000000 +0900
+++ b/cmake/kokkos_enable_devices.cmake	2023-01-27 16:38:58.000000000 +0900
@@ -75,6 +75,7 @@
       Clang      ${ClangOpenMPFlag}
       IntelLLVM  -fiopenmp
       AppleClang -Xpreprocessor -fopenmp
+      FujitsuClang -Kopenmp
       NVHPC      -mp
       Cray       NO-VALUE-SPECIFIED
       XL         -qsmp=omp
diff -u -r a/cmake/kokkos_functions.cmake b/cmake/kokkos_functions.cmake
--- a/cmake/kokkos_functions.cmake	2022-09-21 07:17:28.000000000 +0900
+++ b/cmake/kokkos_functions.cmake	2023-01-27 16:39:55.000000000 +0900
@@ -823,7 +823,7 @@
 ENDFUNCTION()
 
 FUNCTION(COMPILER_SPECIFIC_OPTIONS_HELPER)
-  SET(COMPILERS NVIDIA NVHPC XL XLClang DEFAULT Cray Intel Clang AppleClang IntelLLVM GNU HIPCC Fujitsu)
+  SET(COMPILERS NVIDIA NVHPC XL XLClang DEFAULT Cray Intel Clang AppleClang IntelLLVM GNU HIPCC FujitsuClang Fujitsu)
   CMAKE_PARSE_ARGUMENTS(
     PARSE
     "LINK_OPTIONS;COMPILE_OPTIONS;COMPILE_DEFINITIONS;LINK_LIBRARIES"
