diff -u -r a/cmake_modules/SetupCxxFlags.cmake b/cmake_modules/SetupCxxFlags.cmake
--- a/cmake_modules/SetupCxxFlags.cmake	2021-01-19 07:02:42.000000000 +0900
+++ b/cmake_modules/SetupCxxFlags.cmake	2023-01-31 11:37:28.000000000 +0900
@@ -304,6 +304,7 @@
     set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} /W3")
   elseif(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang"
          OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang"
+         OR CMAKE_CXX_COMPILER_ID STREQUAL "FujitsuClang"
          OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU"
          OR CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
     set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} -Wall")
