Binary files a/.git/index and b/.git/index differ
diff -ru a/sysdep/Makedef.LinuxARM-gnu-ompi b/sysdep/Makedef.LinuxARM-gnu-ompi
--- a/sysdep/Makedef.LinuxARM-gnu-ompi	2023-09-06 09:13:23.619134550 +0900
+++ b/sysdep/Makedef.LinuxARM-gnu-ompi	2023-09-08 15:33:48.102400137 +0900
@@ -12,7 +12,7 @@
 MODDIROPT = -J
 
 FFLAGS_CORE = -cpp                                                                          \
-              -std=f2003 -fall-intrinsics -pedantic-errors -fimplicit-none -fmodule-private \
+              -std=f2003 -fall-intrinsics -fallow-argument-mismatch -fimplicit-none -fmodule-private \
               -fconvert=big-endian -frecord-marker=4 -ffree-form -ffree-line-length-none
 
 FFLAGS_FAST  = -O3 $(FFLAGS_CORE)                                       \
