diff -ruN orig/gromacs-2016.3/src/gromacs/simd/impl_arm_neon/impl_arm_neon_simd_float.h spack-src/src/gromacs/simd/impl_arm_neon/impl_arm_neon_simd_float.h
--- orig/gromacs-2016.3/src/gromacs/simd/impl_arm_neon/impl_arm_neon_simd_float.h	2016-07-09 09:55:39.000000000 +0900
+++ spack-src/src/gromacs/simd/impl_arm_neon/impl_arm_neon_simd_float.h	2021-07-20 14:06:37.451452701 +0900
@@ -568,6 +568,7 @@
     };
 }
 
+#if 0
 static inline SimdFInt32 gmx_simdcall
 operator<<(SimdFInt32 a, int n)
 {
@@ -583,6 +584,7 @@
                vshrq_n_s32(a.simdInternal_, n)
     };
 }
+#endif
 
 static inline SimdFInt32 gmx_simdcall
 operator&(SimdFInt32 a, SimdFInt32 b)
diff -ruN orig/gromacs-2016.3/src/gromacs/simd/impl_arm_neon_asimd/impl_arm_neon_asimd_simd_double.h spack-src/src/gromacs/simd/impl_arm_neon_asimd/impl_arm_neon_asimd_simd_double.h
--- orig/gromacs-2016.3/src/gromacs/simd/impl_arm_neon_asimd/impl_arm_neon_asimd_simd_double.h	2016-07-09 09:55:39.000000000 +0900
+++ spack-src/src/gromacs/simd/impl_arm_neon_asimd/impl_arm_neon_asimd_simd_double.h	2021-07-20 14:08:32.701406416 +0900
@@ -531,6 +531,7 @@
     };
 }
 
+#if 0
 static inline SimdDInt32 gmx_simdcall
 operator<<(SimdDInt32 a, int n)
 {
@@ -546,6 +547,7 @@
         vshr_n_s32(a.simdInternal_, n)
     };
 }
+#endif
 
 static inline SimdDInt32 gmx_simdcall
 operator&(SimdDInt32 a, SimdDInt32 b)
