--- a/qe-7.1/external/fox/config/config.guess	2022-06-15 21:33:59.000000000 +0900
+++ b/qe-7.1/external/fox/config/config.guess	2023-01-13 10:56:45.000000000 +0900
@@ -829,7 +829,7 @@
     i*86:Minix:*:*)
 	echo ${UNAME_MACHINE}-pc-minix
 	exit ;;
-    arm*:Linux:*:*)
+    aarch64:Linux:*:*)
 	echo ${UNAME_MACHINE}-unknown-linux-gnu
 	exit ;;
     avr32*:Linux:*:*)
@@ -909,7 +909,7 @@
 	echo powerpc64-unknown-linux-gnu
 	exit ;;
     ppc64le:Linux:*:*)
-        echo powerpc64-unknown-linux-gnu
+	echo powerpc64-unknown-linux-gnu
 	exit ;;
     alpha:Linux:*:*)
 	case `sed -n '/^cpu model/s/^.*: \(.*\)/\1/p' < /proc/cpuinfo` in
