diff -ur a/CBLAS/CMakeLists.txt b/CBLAS/CMakeLists.txt
--- a/CBLAS/CMakeLists.txt	2021-03-26 03:25:15.000000000 +0900
+++ b/CBLAS/CMakeLists.txt	2022-01-07 16:02:35.000000000 +0900
@@ -6,14 +6,12 @@
 # Create a header file cblas.h for the routines called in my C programs
 include(FortranCInterface)
 ## Ensure that the fortran compiler and c compiler specified are compatible
-FortranCInterface_VERIFY()
+#FortranCInterface_VERIFY()
 FortranCInterface_HEADER(${LAPACK_BINARY_DIR}/include/cblas_mangling.h
                          MACRO_NAMESPACE "F77_"
                          SYMBOL_NAMESPACE "F77_")
 if(NOT FortranCInterface_GLOBAL_FOUND OR NOT FortranCInterface_MODULE_FOUND)
-  message(WARNING "Reverting to pre-defined include/lapacke_mangling.h")
-    configure_file(include/lapacke_mangling_with_flags.h.in
-                  ${LAPACK_BINARY_DIR}/include/lapacke_mangling.h)
+  message(WARNING "Reverting to pre-defined include/cblas_mangling.h")
     configure_file(include/cblas_mangling_with_flags.h.in
                  ${LAPACK_BINARY_DIR}/include/cblas_mangling.h)
 endif()
diff -ur a/LAPACKE/CMakeLists.txt b/LAPACKE/CMakeLists.txt
--- a/LAPACKE/CMakeLists.txt	2021-03-26 03:25:15.000000000 +0900
+++ b/LAPACKE/CMakeLists.txt	2022-01-07 16:03:27.000000000 +0900
@@ -6,7 +6,7 @@
 # Create a header file lapacke_mangling.h for the routines called in my C programs
 include(FortranCInterface)
 ## Ensure that the fortran compiler and c compiler specified are compatible
-FortranCInterface_VERIFY()
+#FortranCInterface_VERIFY()
 FortranCInterface_HEADER(${LAPACK_BINARY_DIR}/include/lapacke_mangling.h
   MACRO_NAMESPACE "LAPACK_"
   SYMBOL_NAMESPACE "LAPACK_")
