diff -Naur a/CBLAS/CMakeLists.txt b/CBLAS/CMakeLists.txt
--- a/CBLAS/CMakeLists.txt	2021-03-25 12:25:15.000000000 -0600
+++ b/CBLAS/CMakeLists.txt	2021-09-01 16:27:23.561355382 -0600
@@ -11,9 +11,7 @@
                          MACRO_NAMESPACE "F77_"
                          SYMBOL_NAMESPACE "F77_")
 if(NOT FortranCInterface_GLOBAL_FOUND OR NOT FortranCInterface_MODULE_FOUND)
-  message(WARNING "Reverting to pre-defined include/lapacke_mangling.h")
-    configure_file(include/lapacke_mangling_with_flags.h.in
-                  ${LAPACK_BINARY_DIR}/include/lapacke_mangling.h)
+  message(WARNING "Reverting to pre-defined include/cblas_mangling.h")
     configure_file(include/cblas_mangling_with_flags.h.in
                  ${LAPACK_BINARY_DIR}/include/cblas_mangling.h)
 endif()
