diff -ur a/CBLAS/CMakeLists.txt b/CBLAS/CMakeLists.txt
--- a/CBLAS/CMakeLists.txt	2022-04-13 05:19:52.000000000 +0900
+++ b/CBLAS/CMakeLists.txt	2023-01-06 17:27:17.000000000 +0900
@@ -6,7 +6,7 @@
 # Create a header file cblas.h for the routines called in my C programs
 include(FortranCInterface)
 ## Ensure that the fortran compiler and c compiler specified are compatible
-FortranCInterface_VERIFY()
+#FortranCInterface_VERIFY()
 FortranCInterface_HEADER(${LAPACK_BINARY_DIR}/include/cblas_mangling.h
                          MACRO_NAMESPACE "F77_"
                          SYMBOL_NAMESPACE "F77_")
diff -ur a/LAPACKE/CMakeLists.txt b/LAPACKE/CMakeLists.txt
--- a/LAPACKE/CMakeLists.txt	2022-04-13 05:19:52.000000000 +0900
+++ b/LAPACKE/CMakeLists.txt	2023-01-06 17:27:54.000000000 +0900
@@ -1,7 +1,7 @@
 # Create a header file lapacke_mangling.h for the routines called in my C programs
 include(FortranCInterface)
 ## Ensure that the fortran compiler and c compiler specified are compatible
-FortranCInterface_VERIFY()
+#FortranCInterface_VERIFY()
 FortranCInterface_HEADER(${LAPACK_BINARY_DIR}/include/lapacke_mangling.h
   MACRO_NAMESPACE "LAPACK_"
   SYMBOL_NAMESPACE "LAPACK_")
