diff -urpN a/src/USER-OMP/pair_lj_cut_omp.cpp b/src/USER-OMP/pair_lj_cut_omp.cpp
--- a/src/USER-OMP/pair_lj_cut_omp.cpp	2020-10-29 08:53:36.000000000 +0900
+++ b/src/USER-OMP/pair_lj_cut_omp.cpp	2021-09-13 18:29:58.000000000 +0900
@@ -109,6 +109,7 @@ void PairLJCutOMP::eval(int iifrom, int
     jnum = numneigh[i];
     fxtmp=fytmp=fztmp=0.0;
 
+    #pragma clang loop vectorize(assume_safety)
     for (jj = 0; jj < jnum; jj++) {
       j = jlist[jj];
       factor_lj = special_lj[sbmask(j)];
