diff --git a/exts/dbcsr/src/dist/dbcsr_dist_methods.F b/exts/dbcsr/src/dist/dbcsr_dist_methods.F
index 38a5724e3..a3f242da4 100644
--- a/exts/dbcsr/src/dist/dbcsr_dist_methods.F
+++ b/exts/dbcsr/src/dist/dbcsr_dist_methods.F
@@ -77,7 +77,7 @@ CONTAINS
          OPTIONAL                                        :: template
       INTEGER, INTENT(IN), OPTIONAL                      :: group
       INTEGER, DIMENSION(:, :), OPTIONAL, POINTER        :: pgrid
-      INTEGER, DIMENSION(:), INTENT(IN), POINTER, CONTIGUOUS :: row_dist, col_dist
+      INTEGER, DIMENSION(:), POINTER, CONTIGUOUS             :: row_dist, col_dist
       LOGICAL, INTENT(IN), OPTIONAL                      :: reuse_arrays

       TYPE(dbcsr_mp_obj)                                 :: mp_env
@@ -116,7 +116,7 @@ CONTAINS
          !! distribution
       TYPE(dbcsr_mp_obj), INTENT(IN)                     :: mp_env
          !! multiprocessing environment
-      TYPE(array_i1d_obj), INTENT(IN)                    :: row_dist_block, col_dist_block
+      TYPE(array_i1d_obj)                                :: row_dist_block, col_dist_block
       TYPE(array_i1d_obj), INTENT(IN), OPTIONAL          :: local_rows, local_cols

       INTEGER, DIMENSION(:), POINTER, CONTIGUOUS         :: cont_row_dist, cont_col_dist, &
@@ -145,7 +145,7 @@ CONTAINS
          !! distribution
       TYPE(dbcsr_mp_obj), INTENT(IN)                     :: mp_env
          !! multiprocessing environment
-      INTEGER, DIMENSION(:), INTENT(IN), POINTER, CONTIGUOUS :: row_dist_block, col_dist_block
+      INTEGER, DIMENSION(:), POINTER, CONTIGUOUS             :: row_dist_block, col_dist_block
       INTEGER, DIMENSION(:), INTENT(IN), OPTIONAL, &
          POINTER, CONTIGUOUS                             :: local_rows, local_cols
       LOGICAL, OPTIONAL

--- orig/cp2k-8.1/src/mp2_eri_ori.f	2022-04-17 21:16:55.000000000 +0900
+++ spack-src/src/mp2_eri.F	2022-04-17 21:17:30.000000000 +0900
@@ -1,3 +1,4 @@
+!options -O1
 !--------------------------------------------------------------------------------------------------!
 !   CP2K: A general program to perform molecular dynamics simulations                              !
 !   Copyright (C) 2000 - 2019  CP2K developers group                                               !
