diff -ur a/blitz/bzconfig.h b/blitz/bzconfig.h
--- a/blitz/bzconfig.h	2019-10-02 07:50:17.000000000 +0900
+++ b/blitz/bzconfig.h	2023-03-28 16:10:57.000000000 +0900
@@ -77,6 +77,10 @@
 /* Pathscale pathCC compiler */
 #include <blitz/pathscale/bzconfig.h>
 
+#elif defined(__FUJITSU)
+/* Fujitsu FCC compiler */
+#include <blitz/fujitsu/bzconfig.h>
+
 #elif defined(__clang__)
 /* clang compiler */
 #include <blitz/llvm/bzconfig.h>
@@ -93,10 +97,6 @@
 /* KAI KCC compiler */
 #include <blitz/kai/bzconfig.h>
 
-#elif defined(__FUJITSU)
-/* Fujitsu FCC compiler */
-#include <blitz/fujitsu/bzconfig.h>
-
 /* Add other compilers here */
 
 #else
diff -ur a/configure b/configure
--- a/configure	2019-10-02 07:50:17.000000000 +0900
+++ b/configure	2023-03-28 15:44:48.000000000 +0900
@@ -5986,6 +5986,8 @@
     COMPILER_VENDOR="compaq" ;; #(
   *aCC*) :
     COMPILER_VENDOR="hp" ;; #(
+  *FCC*) :
+    COMPILER_VENDOR="fujitsu" ;; #(
   *g++*|*c++*) :
     case $target in #(
   *apple*) :
Only in b: configure~
