diff -ruN orig/tau-2.30.2/include/Profile/fujitsu.h tau-2.30.2/include/Profile/fujitsu.h
--- orig/tau-2.30.2/include/Profile/fujitsu.h	2020-11-18 07:10:08.000000000 +0900
+++ tau-2.30.2/include/Profile/fujitsu.h	2022-06-20 17:40:13.447409574 +0900
@@ -13,8 +13,6 @@
 extern char *optarg;
 extern int optind, opterr, optopt;
 
-extern int strcasecmp(const char *s1, const char *s2);
-int strncasecmp(const char *s1, const char *s2, size_t n);
 #ifdef __cplusplus
 }
 #endif /* __cplusplus */
