diff -ruN orig/xmlf90-1.5.2/src/Makefile.am xmlf90-1.5.2/src/Makefile.am
--- orig/xmlf90-1.5.2/src/Makefile.am	2017-12-04 07:14:11.000000000 +0900
+++ xmlf90-1.5.2/src/Makefile.am	2021-08-17 20:49:07.000000000 +0900
@@ -169,7 +169,7 @@
 f03moddir = $(includedir)
 install-data-local:
 	$(INSTALL) -d -m 755 $(DESTDIR)$(f03moddir)
-	$(INSTALL) -m 644 $(x90_f03_ext_mods) $(DESTDIR)$(f03moddir)
+	$(INSTALL) -m 644 $(x90_f03_ext_mods) $(x90_f03_int_mods) $(DESTDIR)$(f03moddir)
 
 uninstall-local:
 	cd $(DESTDIR)$(f03moddir) && rm -f $(x90_f03_ext_mods)
