diff -ruN orig/gromacs-2021.1/computed_checksum gromacs-2021.1-sizeless/computed_checksum
--- orig/gromacs-2021.1/computed_checksum	1970-01-01 09:00:00.000000000 +0900
+++ gromacs-2021.1-sizeless/computed_checksum	2021-05-06 12:51:02.758093930 +0900
@@ -0,0 +1 @@
+99dae498b9a7387131e4e517cb7d868f676e9b2a27be9e89f81ccfdc4074f069
\ No newline at end of file
diff -ruN orig/gromacs-2021.1/src/gromacs/simd/impl_arm_sve/impl_arm_sve_definitions.h gromacs-2021.1-sizeless/src/gromacs/simd/impl_arm_sve/impl_arm_sve_definitions.h
--- orig/gromacs-2021.1/src/gromacs/simd/impl_arm_sve/impl_arm_sve_definitions.h	2021-05-06 16:50:16.621848284 +0900
+++ gromacs-2021.1-sizeless/src/gromacs/simd/impl_arm_sve/impl_arm_sve_definitions.h	2021-05-06 12:41:43.077332393 +0900
@@ -47,8 +47,10 @@
 #define GMX_SIMD 1
 #define GMX_SIMD_HAVE_FLOAT 1
 #define GMX_SIMD_HAVE_FLOAT_ARRAY GMX_SIMD_ARM_SVE_FIXED_VECTOR_LENGTH
+#define GMX_SIMD_HAVE_FLOAT_GLOBAL GMX_SIMD_ARM_SVE_FIXED_VECTOR_LENGTH
 #define GMX_SIMD_HAVE_DOUBLE 1
 #define GMX_SIMD_HAVE_DOUBLE_ARRAY GMX_SIMD_ARM_SVE_FIXED_VECTOR_LENGTH
+#define GMX_SIMD_HAVE_DOUBLE_GLOBAL GMX_SIMD_ARM_SVE_FIXED_VECTOR_LENGTH
 #define GMX_SIMD_HAVE_LOADU 1
 #define GMX_SIMD_HAVE_STOREU 1
 #define GMX_SIMD_HAVE_LOGICAL 1
@@ -56,9 +58,11 @@
 #define GMX_SIMD_HAVE_FINT32_EXTRACT 1
 #define GMX_SIMD_HAVE_FINT32_LOGICAL 1
 #define GMX_SIMD_HAVE_FINT32_ARITHMETICS 1
+#define GMX_SIMD_HAVE_FINT32_GLOBAL GMX_SIMD_ARM_SVE_FIXED_VECTOR_LENGTH
 #define GMX_SIMD_HAVE_DINT32_EXTRACT 1
 #define GMX_SIMD_HAVE_DINT32_LOGICAL 1
 #define GMX_SIMD_HAVE_DINT32_ARITHMETICS 1
+#define GMX_SIMD_HAVE_DINT32_GLOBAL GMX_SIMD_ARM_SVE_FIXED_VECTOR_LENGTH
 #define GMX_SIMD_HAVE_NATIVE_COPYSIGN_FLOAT 0
 #define GMX_SIMD_HAVE_NATIVE_RSQRT_ITER_FLOAT \
     0 // Although there is support, it is disabled in GROMACS, because rsqrtIter does not work correctly for inputs near MAX_FLOAT
@@ -99,11 +103,14 @@
 
 #define GMX_SIMD4_HAVE_FLOAT 1
 #define GMX_SIMD4_HAVE_FLOAT_ARRAY 1
+#define GMX_SIMD4_HAVE_FLOAT_GLOBAL 1
 #if GMX_SIMD_DOUBLE_WIDTH < 4
 #    define GMX_SIMD4_HAVE_DOUBLE 0
 #    define GMX_SIMD4_HAVE_DOUBLE_ARRAY 0
+#    define GMX_SIMD4_HAVE_DOUBLE_GLOBAL 0
 #else
 #    define GMX_SIMD4_HAVE_DOUBLE 1
 #    define GMX_SIMD4_HAVE_DOUBLE_ARRAY GMX_SIMD_ARM_SVE_FIXED_VECTOR_LENGTH
+#    define GMX_SIMD4_HAVE_DOUBLE_GLOBAL GMX_SIMD_ARM_SVE_FIXED_VECTOR_LENGTH
 #endif
 #endif // GMX_SIMD_IMPL_ARM_SVE_DEFINITIONS_H
