diff -ruN orig/gromacs-2022/src/gromacs/tools/tune_pme.cpp gromacs-2022/src/gromacs/tools/tune_pme.cpp
--- orig/gromacs-2022/src/gromacs/tools/tune_pme.cpp	2022-02-23 01:05:08.000000000 +0900
+++ gromacs-2022/src/gromacs/tools/tune_pme.cpp	2022-03-31 14:41:38.034671539 +0900
@@ -709,7 +709,7 @@
         snew(command,
              std::strlen(cmd_mpirun) + std::strlen(cmd_np) + std::strlen(cmd_mdrun)
                      + std::strlen(filename) + 50);
-        sprintf(command, "%s%s%s -version -maxh 0.001 1> %s 2>&1", cmd_mpirun, cmd_np, cmd_mdrun, filename);
+        sprintf(command, "%s%s -std %s %s -version -maxh 0.001", cmd_mpirun, cmd_np, filename, cmd_mdrun);
     }
     fprintf(stdout, "Trying '%s' ... ", command);
     make_backup(filename);
