diff -ru orig/cp2k-7.1/src/eri_mme/eri_mme_error_control.F spack-src/src/eri_mme/eri_mme_error_control.F
--- orig/cp2k-7.1/src/eri_mme/eri_mme_error_control.F	2019-12-24 05:11:56.000000000 +0900
+++ spack-src/src/eri_mme/eri_mme_error_control.F	2021-10-27 11:24:30.562527831 +0900
@@ -295,7 +295,7 @@
       REAL(KIND=dp), DIMENSION(3, 3), INTENT(IN)         :: h_inv
       REAL(KIND=dp), INTENT(IN)                          :: G_min, zet_max
       INTEGER, INTENT(IN)                                :: l_max_zet, n_minimax
-      REAL(KIND=dp), DIMENSION(:), INTENT(OUT)           :: minimax_aw
+      REAL(KIND=dp), DIMENSION(:), INTENT(IN)            :: minimax_aw
       REAL(KIND=dp), INTENT(OUT)                         :: err_ctff, C_mm
       TYPE(cp_para_env_type), INTENT(IN), POINTER        :: para_env
 
diff -ru orig/cp2k-7.1/src/libint_wrapper.F spack-src/src/libint_wrapper.F
--- orig/cp2k-7.1/src/libint_wrapper.F	2019-12-24 05:11:56.000000000 +0900
+++ spack-src/src/libint_wrapper.F	2021-10-27 11:11:35.789219870 +0900
@@ -30,8 +30,8 @@
         libint2_build, libint2_build_eri, libint2_build_eri1, libint2_cleanup_eri, &
         libint2_cleanup_eri1, libint2_init_eri, libint2_init_eri1, libint2_static_cleanup, &
         libint2_static_init, libint_t, libint2_max_am_eri, libint2_init_3eri, libint2_cleanup_3eri, &
-        libint2_init_2eri, libint2_cleanup_2eri!,&
-        !libint2_build_3eri
+        libint2_init_2eri, libint2_cleanup_2eri,&
+        libint2_build_2eri, libint2_build_3eri
 #endif
    USE orbital_pointers,                ONLY: nco
 #include "./base/base_uses.f90"
@@ -76,10 +76,10 @@
    !dimensions of the following array are mixed up. This should be fixed from libint 2.6 on.
    !Then, one will be able to simply USE libint2_build_3eri & libint2_build_2eri, as currently
    !commented above
-   TYPE(C_FUNPTR), DIMENSION(0:LIBINT2_MAX_AM,0:LIBINT2_MAX_AM,0:LIBINT2_MAX_AM_3eri), &
-      BIND(C) :: libint2_build_3eri
-   TYPE(C_FUNPTR), DIMENSION(0:LIBINT2_MAX_AM_2eri, 0:LIBINT2_MAX_AM_2eri), &
-      BIND(C) :: libint2_build_2eri
+   !TYPE(C_FUNPTR), DIMENSION(0:LIBINT2_MAX_AM,0:LIBINT2_MAX_AM,0:LIBINT2_MAX_AM_3eri), &
+      !BIND(C) :: libint2_build_3eri
+   !TYPE(C_FUNPTR), DIMENSION(0:LIBINT2_MAX_AM_2eri, 0:LIBINT2_MAX_AM_2eri), &
+      !BIND(C) :: libint2_build_2eri
 #endif
 
 CONTAINS
