--- modified	2024-11-12 02:06:17.155243503 +0100
+++ lib/common/compiler.h	2024-03-21 23:05:51.000000000 +0100
@@ -181,10 +181,19 @@
  * If you can remove a LIKELY/UNLIKELY annotation without speed changes in gcc
  * and clang, please do.
  */
+#if defined(__GNUC__)
+#define LIKELY(x) (__builtin_expect((x), 1))
+#define UNLIKELY(x) (__builtin_expect((x), 0))
+#else
 #define LIKELY(x) (x)
 #define UNLIKELY(x) (x)
+#endif
 
-#define ZSTD_UNREACHABLE do { assert(0); } while (0)
+#if __has_builtin(__builtin_unreachable) || (defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 5)))
+#  define ZSTD_UNREACHABLE do { assert(0), __builtin_unreachable(); } while (0)
+#else
+#  define ZSTD_UNREACHABLE do { assert(0); } while (0)
+#endif
 
 /* disable warnings */
 #ifdef _MSC_VER    /* Visual Studio */
